### ISSUE TITLE
Cargo does not pass `--target` to `rustdoc --test` for the host target

### ISSUE THREAD

**Issue by jyn514** [rust-lang/cargo#10560]
### Problem

`cargo test --doc --target x86_64-unknown-linux-gnu` does not pass a `--target` flag to rustdoc if the host target is x86_64-unknown-linux-gnu. This is inconsistent with `cargo doc/check/build`. It caused trouble for bootstrap in https://github.com/rust-lang/rust/pull/95993.

### Steps

1. `cargo new --lib example`
2. `cd example`
3. `cargo test --doc --target x86_64-unknown-linux-gnu -v`

Observe that no `--target` argument is passed to rustdoc.

### Possible Solution(s)

Pass `--target` to rustdoc.

### Version

```text
cargo 1.61.0-nightly (109bfbd 2022-03-17)
release: 1.61.0-nightly
commit-hash: 109bfbd055325ef87a6e7f63d67da7e838f8300b
commit-date: 2022-03-17
host: x86_64-unknown-linux-gnu
libgit2: 1.4.2 (sys:0.14.2 vendored)
libcurl: 7.80.0-DEV (sys:0.4.51+curl-7.80.0 vendored ssl:OpenSSL/1.1.1m)
os: Pop!_OS 20.04 (focal) [64-bit]
```

**Comment by weihanglo** [comment]
Hmm… I believe the feature [`-Zdoctest-xcompile`] is still unstable. As far as I know, it waits for rustdoc [`--enable-per-target-ignores`] and [`--runtool`|`--runtool-arg`] to be stabilized.

Closing in favor of the tracking issues https://github.com/rust-lang/rust/issues/64245 and https://github.com/rust-lang/cargo/issues/7040. Thanks!

[`-Zdoctest-xcompile`]: https://doc.rust-lang.org/nightly/cargo/reference/unstable.html#doctest-xcompile
[`--enable-per-target-ignores`]: https://doc.rust-lang.org/nightly/rustdoc/unstable-features.html#--enable-per-target-ignores-allow-ignore-foo-style-filters-for-doctests
[`--runtool`|`--runtool-arg`]: https://doc.rust-lang.org/nightly/rustdoc/unstable-features.html#--runtool---runtool-arg-program-to-run-tests-with-args-to-pass-to-it

**Comment by jyn514** [comment]
@weihanglo no, this does not depend on -Zdoctest-xcompile, it works on stable.

**Comment by jyn514** [comment]
Also, this is about `cargo doc`, not `cargo test --doc`.

**Comment by weihanglo** [comment]
Sorry. Did you mean Cargo should pass `--target` to rustdoc for the host target only?

**Comment by jyn514** [comment]
Yes, exactly. That matches the behavior of `cargo build` and `cargo doc`.
